oap-server/server-library/library-util/src/main/java/org/apache/skywalking/oap/server/library/util/FieldsHelper.java

```diff
@@ -28,51 +38,43 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.BiConsumer;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
-import org.apache.commons.lang3.StringUtils;
-import org.apache.skywalking.oap.server.library.module.ModuleStartException;
-import org.apache.skywalking.oap.server.library.util.ResourceUtils;
-import org.apache.skywalking.oap.server.receiver.envoy.als.ServiceMetaInfo;
-import org.apache.skywalking.oap.server.receiver.envoy.als.ServiceMetaInfo.KeyValue;
-import org.yaml.snakeyaml.Yaml;
-
-import com.google.common.base.Splitter;
-import com.google.common.base.Strings;
-import com.google.protobuf.Struct;
-import com.google.protobuf.Value;
-
-import lombok.RequiredArgsConstructor;
-import lombok.experimental.Delegate;
-import lombok.extern.slf4j.Slf4j;
-
 @Slf4j
-public enum FieldsHelper {
-    SINGLETON;
+public class FieldsHelper {
+    private static final Map<Class<?>, FieldsHelper> HELPER_MAP = new ConcurrentHashMap<>();
```

Comment:
Could you explain this? It seems to you expand the scope of this class to cover more field processes?

Reply:
Updated. Yes, Before we need to use `FieldsHelper.SINGLE` for getting the instance, but it only works for the ALS message. But for now, the cilium fetcher still needs this for the inflate metadata, so I make it an instance with the target inflate class. When the user needs to get the instance of `FieldsHelper` needs to call `forClass(Class)`.